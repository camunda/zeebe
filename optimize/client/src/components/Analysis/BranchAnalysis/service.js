/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {post} from 'request';

export async function loadFrequencyData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  identifier,
  filter
) {
  const response = await post(
    'api/report/evaluate',
    createFlowNodeFrequencyReport(
      processDefinitionKey,
      processDefinitionVersions,
      tenantIds,
      identifier,
      filter
    )
  );

  return await response.json();
}

function createFlowNodeFrequencyReport(key, versions, tenantIds, identifier, filter) {
  return {
    combined: false,
    reportType: 'process',
    data: {
      definitions: [{key, versions, tenantIds, identifier}],
      filter,
      view: {
        entity: 'flowNode',
        properties: ['frequency'],
      },
      groupBy: {
        type: 'flowNodes',
        unit: null,
      },
      visualization: 'heat',
    },
  };
}

export async function loadCorrelationData(
  processDefinitionKey,
  processDefinitionVersions,
  tenantIds,
  filter,
  gateway,
  end
) {
  const response = await post('api/analysis/correlation', {
    processDefinitionKey,
    processDefinitionVersions,
    tenantIds,
    filter,
    gateway,
    end,
  });

  return await response.json();
}
