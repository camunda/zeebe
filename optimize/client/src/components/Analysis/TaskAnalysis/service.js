/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {post} from 'request';

export async function loadNodesOutliers(config) {
  const response = await post('api/analysis/flowNodeOutliers', config);
  return await response.json();
}

export {
  loadDurationData,
  loadCommonOutliersVariables,
  getOutlierSummary,
  shouldUseLogharitmicScale,
} from './service.ts';
