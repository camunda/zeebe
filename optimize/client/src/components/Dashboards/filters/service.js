/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {post, get} from 'request';

export async function getVariableNames(reportIds) {
  const response = await post('api/variables/reports', {reportIds});

  return await response.json();
}

export async function getVariableValues(reportIds, name, type, numResults, valueFilter) {
  const response = await post('api/variables/values/reports', {
    reportIds,
    name,
    type,
    valueFilter,
    numResults,
    resultOffset: 0,
  });

  return await response.json();
}

export async function loadUsersByReportIds(type, payload) {
  const response = await post(`api/${type}/search/reports`, payload);

  return await response.json();
}

export async function getAssigneeNames(type, listOfIds) {
  const response = await get('api/' + type, {idIn: listOfIds.join(',')});

  return await response.json();
}

export function isOfType(filter, availableFilter) {
  switch (filter.type) {
    case 'runningInstancesOnly':
    case 'completedInstancesOnly':
    case 'canceledInstancesOnly':
    case 'nonCanceledInstancesOnly':
    case 'suspendedInstancesOnly':
    case 'nonSuspendedInstancesOnly':
      return availableFilter.type === 'state';
    case 'instanceStartDate':
    case 'instanceEndDate':
    case 'assignee':
    case 'candidateGroup':
      return availableFilter.type === filter.type;
    case 'variable':
      return (
        availableFilter.type === 'variable' &&
        availableFilter.data.type === filter.data.type &&
        availableFilter.data.name === filter.data.name
      );
    default:
      return false;
  }
}
