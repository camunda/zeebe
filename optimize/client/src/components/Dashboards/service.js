/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get, del, post} from 'request';

export async function shareDashboard(dashboardId) {
  const body = {
    dashboardId,
  };
  const response = await post(`api/share/dashboard`, body);

  const json = await response.json();
  return json.id;
}

export async function getSharedDashboard(reportId) {
  const response = await get(`api/share/dashboard/${reportId}`);

  if (response.status > 201) {
    return '';
  } else {
    const json = await response.json();
    return json.id;
  }
}

export async function revokeDashboardSharing(id) {
  return await del(`api/share/dashboard/${id}`);
}

export async function isAuthorizedToShareDashboard(dashboardId) {
  try {
    const response = await get(`api/share/dashboard/${dashboardId}/isAuthorizedToShare`);
    return response.status === 200;
  } catch (error) {
    return false;
  }
}

export function convertFilterToDefaultValues(availableFilter, filters) {
  // dates
  if (availableFilter.type === 'instanceStartDate' || availableFilter.type === 'instanceEndDate') {
    return filters.find(({type}) => type === availableFilter.type)?.data ?? null;
  }

  // assignee / candidate group
  if (availableFilter.type === 'assignee' || availableFilter.type === 'candidateGroup') {
    return (
      filters.find(
        ({type, data}) =>
          type === availableFilter.type && data.operator === availableFilter.data.operator
      )?.data.values ?? null
    );
  }

  // state
  if (availableFilter.type === 'state') {
    const defaultValues = filters
      .filter(({type}) =>
        [
          'runningInstancesOnly',
          'completedInstancesOnly',
          'canceledInstancesOnly',
          'nonCanceledInstancesOnly',
          'suspendedInstancesOnly',
          'nonSuspendedInstancesOnly',
        ].includes(type)
      )
      .map(({type}) => type);

    if (defaultValues.length > 0) {
      return defaultValues;
    }
    return null;
  }

  // Boolean Variables
  if (availableFilter.type === 'variable' && availableFilter.data.type === 'Boolean') {
    return (
      filters.find(
        ({type, data}) =>
          type === 'variable' && data.type === 'Boolean' && data.name === availableFilter.data.name
      )?.data.data.values ?? null
    );
  }

  // Date Variables
  if (availableFilter.type === 'variable' && availableFilter.data.type === 'Date') {
    return (
      filters.find(
        ({type, data}) =>
          type === 'variable' && data.type === 'Date' && data.name === availableFilter.data.name
      )?.data.data ?? null
    );
  }

  // string and number variables
  if (availableFilter.type === 'variable') {
    return (
      filters.find(
        ({type, data}) =>
          type === 'variable' &&
          data.type === availableFilter.data.type &&
          data.name === availableFilter.data.name &&
          data.data.operator === availableFilter.data.data.operator
      )?.data.data.values ?? null
    );
  }

  return null;
}

export function getDefaultFilter(availableFilters) {
  const filters = [];

  availableFilters.forEach((availableFilter) => {
    if (availableFilter?.data?.defaultValues) {
      if (availableFilter.type === 'state') {
        return filters.push(
          ...availableFilter.data.defaultValues.map((type) => ({
            filterLevel: 'instance',
            type,
            data: null,
          }))
        );
      }

      if (
        availableFilter.type === 'instanceStartDate' ||
        availableFilter.type === 'instanceEndDate'
      ) {
        return filters.push({
          type: availableFilter.type,
          filterLevel: 'instance',
          data: availableFilter.data.defaultValues,
        });
      }

      if (availableFilter.type === 'assignee' || availableFilter.type === 'candidateGroup') {
        return filters.push({
          type: availableFilter.type,
          filterLevel: 'view',
          data: {
            operator: availableFilter.data.operator,
            values: availableFilter.data.defaultValues,
          },
        });
      }

      if (availableFilter.type === 'variable' && availableFilter.data.type === 'Boolean') {
        return filters.push({
          type: availableFilter.type,
          filterLevel: 'instance',
          data: {
            name: availableFilter.data.name,
            type: availableFilter.data.type,
            data: {values: availableFilter.data.defaultValues},
          },
        });
      }

      if (availableFilter.type === 'variable' && availableFilter.data.type === 'Date') {
        return filters.push({
          type: availableFilter.type,
          filterLevel: 'instance',
          data: {
            name: availableFilter.data.name,
            type: availableFilter.data.type,
            data: availableFilter.data.defaultValues,
          },
        });
      }

      if (availableFilter.type === 'variable') {
        return filters.push({
          type: availableFilter.type,
          filterLevel: 'instance',
          data: {
            name: availableFilter.data.name,
            type: availableFilter.data.type,
            data: {
              operator: availableFilter.data.data.operator,
              values: availableFilter.data.defaultValues,
            },
          },
        });
      }
    }
  });

  return filters;
}
