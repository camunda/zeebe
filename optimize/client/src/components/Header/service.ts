/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get} from 'request';

export async function isEventBasedProcessEnabled(): Promise<boolean> {
  const response = await get('api/eventBasedProcess/isEnabled');
  const value = await response.text();

  return value === 'true';
}

export async function getUserToken(): Promise<string> {
  const response = await get('api/token');
  const value = (await response.json()) as {token: string};
  return value.token;
}
