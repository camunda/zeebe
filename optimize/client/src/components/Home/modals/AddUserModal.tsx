/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {useState} from 'react';
import {Button, Form, RadioButton, RadioButtonGroup, Stack} from '@carbon/react';

import {Modal, UserTypeahead, User} from 'components';
import {getOptimizeProfile} from 'config';
import {t} from 'translation';

interface Role extends User {
  role: 'viewer' | 'editor' | 'manager';
}

interface AddUserModalProps {
  optimizeProfile: Awaited<ReturnType<typeof getOptimizeProfile>>;
  open: boolean;
  existingUsers: Role[];
  onConfirm: (users: Omit<User, 'id'>[]) => void;
  onClose: () => void;
}

export default function AddUserModal(props: AddUserModalProps) {
  const [users, setUsers] = useState<User[]>([]);
  const [activeRole, setActiveRole] = useState<Role['role']>('viewer');

  const onConfirm = () => {
    if (!users.length) {
      return;
    }

    props.onConfirm(users.map(({identity}) => ({role: activeRole, identity})));
    reset();
  };

  const onClose = () => {
    props.onClose();
    reset();
  };

  const reset = () => {
    setUsers([]);
    setActiveRole('viewer');
  };

  const {optimizeProfile, open, existingUsers} = props;

  return (
    <Modal className="AddUserModal" open={open} onClose={onClose} isOverflowVisible>
      <Modal.Header
        title={
          optimizeProfile === 'platform'
            ? t('home.roles.addUserGroupTitle')
            : t('home.roles.addUserTitle')
        }
      />
      <Modal.Content>
        <Form>
          <Stack gap={6}>
            <UserTypeahead
              titleText={
                optimizeProfile === 'platform' ? t('home.userGroupsTitle') : t('home.userTitle')
              }
              users={users}
              collectionUsers={existingUsers}
              onChange={(users: User[]) => setUsers(users)}
              optionsOnly={optimizeProfile === 'cloud'}
            />
            <RadioButtonGroup
              name="userRole"
              legendText={t('home.roles.userRole')}
              orientation="vertical"
            >
              <RadioButton
                value="viewer"
                checked={activeRole === 'viewer'}
                onClick={() => setActiveRole('viewer')}
                labelText={
                  <>
                    <span>{t('home.roles.viewer')}</span>
                    <span className="subtitle">{t('home.roles.viewer-description')}</span>
                  </>
                }
              />
              <RadioButton
                value="editor"
                checked={activeRole === 'editor'}
                onClick={(a) => setActiveRole('editor')}
                labelText={
                  <>
                    <span>{t('home.roles.editor')}</span>
                    <span className="subtitle">{t('home.roles.editor-description')}</span>
                  </>
                }
              />
              <RadioButton
                value="manager"
                checked={activeRole === 'manager'}
                onClick={() => setActiveRole('manager')}
                labelText={
                  <>
                    <span>{t('home.roles.manager')}</span>
                    <span className="subtitle">{t('home.roles.manager-description')}</span>
                  </>
                }
              />
            </RadioButtonGroup>
          </Stack>
        </Form>
      </Modal.Content>
      <Modal.Footer>
        <Button kind="secondary" className="cancel" onClick={onClose}>
          {t('common.cancel')}
        </Button>
        <Button className="confirm" disabled={!users.length} onClick={onConfirm}>
          {t('common.add')}
        </Button>
      </Modal.Footer>
    </Modal>
  );
}
