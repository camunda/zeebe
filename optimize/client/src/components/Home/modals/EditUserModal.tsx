/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {useState} from 'react';
import {Button, RadioButton, Form, RadioButtonGroup} from '@carbon/react';

import {Modal} from 'components';
import {t} from 'translation';

interface EditUserModalProps {
  identity: {name: string; id: string};
  initialRole: string;
  onClose: () => void;
  onConfirm: (name: string) => void;
}

export default function EditUserModal({
  identity,
  initialRole,
  onClose,
  onConfirm,
}: EditUserModalProps) {
  const {name, id} = identity;
  const [role, setRole] = useState(initialRole);

  return (
    <Modal className="EditUserModal" open onClose={onClose}>
      <Modal.Header title={t('common.editName', {name: name || id})} />
      <Modal.Content>
        <Form>
          <RadioButtonGroup
            name="userRole"
            legendText={t('home.roles.userRole')}
            orientation="vertical"
          >
            <RadioButton
              value="viewer"
              checked={role === 'viewer'}
              onClick={() => setRole('viewer')}
              labelText={
                <>
                  <span>{t('home.roles.viewer')}</span>
                  <span className="subtitle">{t('home.roles.viewer-description')}</span>
                </>
              }
              type="radio"
            />
            <RadioButton
              value="editor"
              checked={role === 'editor'}
              onClick={() => setRole('editor')}
              labelText={
                <>
                  <span>{t('home.roles.editor')}</span>
                  <span className="subtitle">{t('home.roles.editor-description')}</span>
                </>
              }
              type="radio"
            />
            <RadioButton
              value="manager"
              checked={role === 'manager'}
              onClick={() => setRole('manager')}
              labelText={
                <>
                  <span>{t('home.roles.manager')}</span>
                  <span className="subtitle">{t('home.roles.manager-description')}</span>
                </>
              }
              type="radio"
            />
          </RadioButtonGroup>
        </Form>
      </Modal.Content>
      <Modal.Footer>
        <Button kind="secondary" className="cancel" onClick={onClose}>
          {t('common.cancel')}
        </Button>
        <Button className="confirm" onClick={() => onConfirm(role)}>
          {t('common.apply')}
        </Button>
      </Modal.Footer>
    </Modal>
  );
}
