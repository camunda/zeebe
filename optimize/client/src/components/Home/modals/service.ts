/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {get} from 'request';
import {Definition, Tenant} from 'types';
import {formatters, UNAUTHORIZED_TENANT_ID} from 'services';
import {t} from 'translation';

const {formatTenantName} = formatters;

export type TenantWithDefinitions = Tenant & {definitions: Definition};

export type DefinitionWithTenants = Omit<Definition, 'key'> & {tenants: Tenant[]; key: string};

export async function getDefinitionsWithTenants(): Promise<DefinitionWithTenants[]> {
  const response = await get('api/definition');

  return await response.json();
}

export async function getTenantsWithDefinitions(): Promise<TenantWithDefinitions[]> {
  const response = await get('api/definition/_groupByTenant');

  return await response.json();
}

export async function getDefinitionTenants(
  defintionKey: string,
  defintionType: string
): Promise<DefinitionWithTenants> {
  const response = await get(`api/definition/${defintionType}/${defintionKey}`);

  return await response.json();
}

export function formatTenants(tenants: Tenant[], selectedTenants: Tenant[]) {
  return tenants.map(({id, name}, index) => {
    if (id === UNAUTHORIZED_TENANT_ID) {
      return {
        id: index,
        label: t('home.sources.unauthorizedTenant'),
        checked: true,
        disabled: true,
      };
    }

    return {
      id,
      label: formatTenantName({id, name}),
      checked: selectedTenants.some((tenant) => tenant.id === id),
    };
  });
}
