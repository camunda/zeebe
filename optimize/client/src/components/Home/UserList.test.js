/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {Deleter, EntityList} from 'components';
import {getOptimizeProfile} from 'config';

import {editUser, removeUser} from './service';
import AddUserModal from './modals/AddUserModal';
import EditUserModal from './modals/EditUserModal';

import UserListWithErrorHandling from './UserList';

jest.mock('config', () => ({
  getOptimizeProfile: jest.fn().mockReturnValue('platform'),
}));

jest.mock('./service', () => ({
  getUsers: jest.fn().mockReturnValue([
    {
      id: 'USER:kermit',
      identity: {
        id: 'kermit',
        type: 'user', // or group
      },
      role: 'manager', // or editor, viewer
    },
    {
      id: 'GROUP:sales',
      identity: {
        id: 'sales',
        type: 'group',
        memberCount: 2,
      },
      role: 'manager',
    },
  ]),
  editUser: jest.fn(),
  removeUser: jest.fn(),
}));

const UserList = UserListWithErrorHandling.WrappedComponent;

const props = {
  mightFail: jest.fn().mockImplementation((data, cb) => cb(data)),
  collection: 'collectionId',
  readOnly: false,
  onChange: jest.fn(),
};

it('should match snapshot', async () => {
  const node = await shallow(<UserList {...props} />);

  expect(node).toMatchSnapshot();
});

it('should hide add button and edit menu when in readOnly mode', () => {
  const node = shallow(<UserList {...props} readOnly />);

  expect(node.find(EntityList).prop('action')).toBe(false);

  expect(node.find(EntityList).prop('rows')[0].actions).toBe(false);
});

it('should pass Entity to Deleter', () => {
  const node = shallow(<UserList {...props} />);

  node.find(EntityList).prop('rows')[0].actions[1].action();

  expect(node.find(Deleter).prop('entity').id).toBe('kermit');
});

it('should delete collection', () => {
  const node = shallow(<UserList {...props} />);

  node.setState({deleting: {id: 'USER:kermit', identity: {id: 'kermit'}}});
  node.find(Deleter).prop('deleteEntity')();

  expect(removeUser).toHaveBeenCalledWith('collectionId', 'USER:kermit');
});

it('should show an edit modal when clicking the edit button', () => {
  const node = shallow(<UserList {...props} />);

  node.find(EntityList).prop('rows')[0].actions[0].action();

  expect(node.find(EditUserModal)).toExist();
});

it('should modify the user role', () => {
  const node = shallow(<UserList {...props} />);

  node.setState({editing: {id: 'USER:kermit', identity: {id: 'kermit'}}});
  node.find(EditUserModal).prop('onConfirm')('viewer');

  expect(editUser).toHaveBeenCalledWith('collectionId', 'USER:kermit', 'viewer');
  expect(props.onChange).toHaveBeenCalled();
});

it('should pass optimize environment to addUserModal', async () => {
  getOptimizeProfile.mockReturnValueOnce('ccsm');
  const node = await shallow(<UserList {...props} />);

  node.find(EntityList).prop('action').props.onClick({});

  expect(node.find(AddUserModal).prop('optimizeProfile')).toBe('ccsm');
});

it('should hide members column in ccsm and cloud', async () => {
  getOptimizeProfile.mockReturnValueOnce('ccsm');
  let node = shallow(<UserList {...props} />);

  expect(node.find(EntityList).prop('headers')).toEqual(['Name', 'Role']);
  expect(node.find(EntityList).prop('rows')[0].meta.length).toBe(1);

  getOptimizeProfile.mockReturnValueOnce('platform');
  node = shallow(<UserList {...props} />);

  await flushPromises();

  expect(node.find(EntityList).prop('headers')).toEqual(['Name', 'Members', 'Role']);
  expect(node.find(EntityList).prop('rows')[0].meta.length).toBe(2);

  getOptimizeProfile.mockReturnValueOnce('cloud');
  node = shallow(<UserList {...props} />);

  await flushPromises();

  expect(node.find(EntityList).prop('headers')).toEqual(['Name', 'Role']);
  expect(node.find(EntityList).prop('rows')[0].meta.length).toBe(1);
});
