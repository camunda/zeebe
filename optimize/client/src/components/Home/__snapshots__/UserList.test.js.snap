// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="UserList"
>
  <EntityList
    action={
      <Button
        kind="primary"
        onClick={[Function]}
      >
        Add
      </Button>
    }
    bulkActions={
      Array [
        <BulkDeleter
          deleteEntities={[Function]}
          type="remove"
        />,
      ]
    }
    emptyStateComponent={
      <EmptyState
        icon={<ForwardRef(User) />}
        title="No results found"
      />
    }
    headers={
      Array [
        "Name",
        "Members",
        "Role",
      ]
    }
    isLoading={false}
    onChange={[Function]}
    rows={
      Array [
        Object {
          "actions": Array [
            Object {
              "action": [Function],
              "icon": <ForwardRef(Edit) />,
              "text": "Edit",
            },
            Object {
              "action": [Function],
              "icon": <ForwardRef(TrashCan) />,
              "text": "Remove",
            },
          ],
          "className": "user",
          "entityType": "user",
          "icon": <ForwardRef(User) />,
          "id": "USER:kermit",
          "meta": Array [
            false,
            "Manager",
          ],
          "name": "kermit",
          "type": "User",
        },
        Object {
          "actions": Array [
            Object {
              "action": [Function],
              "icon": <ForwardRef(Edit) />,
              "text": "Edit",
            },
            Object {
              "action": [Function],
              "icon": <ForwardRef(TrashCan) />,
              "text": "Remove",
            },
          ],
          "className": "group",
          "entityType": "user",
          "icon": <ForwardRef(Group) />,
          "id": "GROUP:sales",
          "meta": Array [
            "2 Users",
            "Manager",
          ],
          "name": "sales",
          "type": "User group",
        },
      ]
    }
  />
  <DeleterErrorHandler
    deleteEntity={[Function]}
    descriptionText="The '' null will be removed from the collection."
    onClose={[Function]}
    onDelete={[Function]}
  />
  <AddUserModal
    existingUsers={
      Array [
        Object {
          "id": "USER:kermit",
          "identity": Object {
            "id": "kermit",
            "type": "user",
          },
          "role": "manager",
        },
        Object {
          "id": "GROUP:sales",
          "identity": Object {
            "id": "sales",
            "memberCount": 2,
            "type": "group",
          },
          "role": "manager",
        },
      ]
    }
    onClose={[Function]}
    onConfirm={[Function]}
    open={false}
    optimizeProfile="platform"
  />
</div>
`;
